### PR TITLE
17 Chat system homework #1 - Lachezar Velinov

### PR DESCRIPTION
**Pull request by Lachezar-Velinov**

### CODE REVIEW

**Comment by arnaudoff** [comment]
Please drop the .idea files from the merge request - they are not needed

File: homeworks/17/chat-system/src/main/java/org/example/ClientHandler.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        } else if (messageToSend.startsWith("/rename ")) {
+            broadcastAnnouncement("SERVER: " + username + " has changed their name to " + messageToSend.substring("/rename ".length()));
+            username = messageToSend.substring("/rename ".length());
+        } else if (messageToSend.startsWith("/message ")) {
+
+            for (ClientHandler clientHandler : clientHandlers) {
+                try {
+                    if (!clientHandler.username.equals(username)) {
+                        clientHandler.bufferedWriter.write("[" + DateTimeFormatter.ofPattern("HH:mm a").format(LocalTime.now()) + "] ");
+                        clientHandler.bufferedWriter.write(username + ": " + messageToSend.substring("/message ".length()));
```
Comment (elenavarbanova) [inline homeworks/17/chat-system/src/main/java/org/example/ClientHandler.java:109]: The condition to the task is not complied.
"/rename" -> "/nick"
"/message" -> "/msg"

File: homeworks/17/chat-system/src/main/java/org/example/Server.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    public Server(ServerSocket serverSocket) {
+        setServerSocket(serverSocket);
+    }
+
+    public static void main(String[] args) throws IOException {
+        int port = 42069;
+        ServerSocket serverSocket = new ServerSocket(port);
+        Server server = new Server(serverSocket);
+        System.out.println("Port is " + port + ". Machkai tigre!");
```
Comment (elenavarbanova) [inline homeworks/17/chat-system/src/main/java/org/example/Server.java:21]: I don't think this is appropriate end of message. There is no need to show what port are we on.

File: homeworks/17/chat-system/src/main/java/org/example/Client.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+import java.net.Socket;
+import java.security.InvalidParameterException;
+import java.util.Objects;
+import java.util.Scanner;
+
+public class Client {
+    private Socket socket;
+    private BufferedReader bufferedReader;
+    private BufferedWriter bufferedWriter;
+    private String username = "Anonimen";
```
Comment (elenavarbanova) [inline homeworks/17/chat-system/src/main/java/org/example/Client.java:13]: There should not be a name for a new user. The should name themselves after connection.

File: homeworks/17/chat-system/src/main/java/org/example/ClientHandler.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                        clientHandler.bufferedWriter.newLine();
+                        clientHandler.bufferedWriter.flush();
+                    }
+                } catch (IOException exception) {
+                    closeEverything(this.socket, bufferedReader, bufferedWriter);
+                }
+            }
+        }
+    }
```
Comment (elenavarbanova) [inline homeworks/17/chat-system/src/main/java/org/example/ClientHandler.java:118]: I think it is a good idea if the command is wrong to tell the user which command they can use.

File: homeworks/17/chat-system/src/main/java/org/example/Client.java [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+                        flag = false;
+                    }
+                }
+            }catch (Exception exception) {
+                System.out.println("Invalid Input");
+            }
+        }while(flag);
+
+        Socket socket = new Socket(host, port);
+        Client client = new Client(socket);
```
Comment (elenavarbanova) [inline homeworks/17/chat-system/src/main/java/org/example/Client.java:53]: Socket and client should be created in the do/while cycle while socket is null.

**Review by elenavarbanova — COMMENTED** [review]
Overal your code is readable, but there are some issues. The biggest are the names of the commands, inappropriate ending of messages and connection to predefined port. Based on the review I think your grade should be **4.50/6.00**.

**Pull request merged**